src/app/data-models/flashcard-model.ts

```diff
@@ -0,0 +1,26 @@
+export interface FlashcardModel {
+  term: string;
+  definition: string;
+  /** Checks if the flashcard is valid. Valid flashcards cannot have an empty
+   * term nor definition;
+   * @returns whether flashcard is valid
+   */
+  isValid(): boolean;
+}
+
+export class FlashcardData implements FlashcardModel{
+  public term: string;
+  public definition: string;
+
```

Comment:
I have added a `image` property to flashcards, and it can be either a URL or undefined. There is also a method to safely check if the flashcard has a url.